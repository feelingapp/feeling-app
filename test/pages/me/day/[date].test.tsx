import * as nextRouter from "next/router"
import { mount, shallow } from "enzyme"

import Date from "../../../../src/pages/me/day/[date]"

nextRouter.useRouter = jest.fn()

describe("Me: ", () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it("matches snapshot", () => {
    nextRouter.useRouter.mockImplementation(() => ({
      query: { date: "01-01-2020" },
    }))
    const wrapper = shallow(<Date />)
    expect(wrapper).toMatchSnapshot()
  })

  it("renders nothing when date is not provided", () => {
    nextRouter.useRouter.mockImplementation(() => ({
      query: { date: undefined },
    }))
    const wrapper = shallow(<Date />)
    expect(wrapper).toMatchSnapshot()
  })

  it("redirects to /me when no date is provided", () => {
    const mockPush = jest.fn()
    nextRouter.useRouter.mockImplementation(() => ({
      query: { date: null },
      push: mockPush,
    }))
    mount(<Date />)
    expect(mockPush).toHaveBeenCalledWith("/me")
  })
})

it("Renders nothing when date is not provided", () => {
  nextRouter.useRouter.mockImplementation(() => ({
    query: { date: null },
  }))
  const wrapper = shallow(<Date />)
  expect(wrapper).toMatchSnapshot()
})

it("Redirects to /me when no date is provided", () => {
  const mockPush = jest.fn()
  nextRouter.useRouter.mockImplementation(() => ({
    query: { date: null },
    push: mockPush,
  }))
  const wrapper = mount(<Date />)
  expect(mockPush).toHaveBeenCalledWith("/me")
})
