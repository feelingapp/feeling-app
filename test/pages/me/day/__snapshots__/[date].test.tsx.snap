// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Me:  matches snapshot 1`] = `
<Fragment>
  <Head>
    <title>
      1st January 2020
       / Feeling
    </title>
  </Head>
  <TopBar
    backButton={true}
    title="1st January 2020"
  />
  <Container />
</Fragment>
`;

exports[`Me:  renders nothing when date is not provided 1`] = `""`;

exports[`Renders nothing when date is not provided 1`] = `""`;
